openapi/v2/Bom/bom-get.yml

```diff
@@ -0,0 +1,209 @@
+openapi: 3.0.0
+info:
+  version: '3.0'
+  title: Get Bom API
+  license:
+    name: MIT
+servers:
+- url: https://{orgId}.salesforce.com/services/apexrest/{namespace}/v2
+  variables:
+    namespace:
+      default: PDLM
+      description: This value should default to PDLM for production environments.
+        For dev environments, provide the namespace to your dev org.
+paths:
+  '/item/{itemNumber}/bom':
+    get:
+      summary: BOM
+      description: Query to return a list of assembly records which represent the full BOM for the
+        given Item.
+      operationId: getBom
+      tags:
+      - Bill of Material
+      - BOM
+      - Assembly
+      parameters:
+      - name: itemNumber
+        in: path
+        description: Item Revision Id or Item Number of assembly to retrieve the BOM for. If Item Id or Item Number is provided, will default to latest released revision or latest revision if there is no released revision.
+        required: true
+        schema:
+          type: string
+          format: string
+      - name: level
+        in: query
+        description: Level of BOM to fetch. Defaults to All levels.
+        required: false
+        schema:
+          type: integer
+          format: integer
+          default: 0
+      - name: delta
+        in: query
+        description: Returns the deltas in the BOM between the latest released revision and previous revision in addition to the BOM.
+        required: false
+        schema:
+          type: boolean
+          format: boolean
+          default: false
+      - name: flatten
+        in: query
+        description: Returns a list of parts in the assembly without level. Will combine duplicate BOM rows that exist at different levels into a single BOM row.
+        required: false
+        schema:
+          type: boolean
+          format: boolean
+          default: false
+      responses:
+        '200':
+          description: Bill of Materials for a specified Item
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetBom'
+        '404':
+          description: Change Order not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+components:
+  schemas:
+    GetBom:
+      type: object
+      required:
+      - returnBom
+      properties:
+        returnBom:
+          type: array
+          items:
+            $ref: '#/components/schemas/ReturnBom'
+          description: Array of objects containing information about the manufacturer items and assemblies associated with the bom.
+        deltaBom:
+          type: object
+          items:
+            $ref: '#/components/schemas/DeltaBom'
+          description: Contains information about changes in the bom.
+    ReturnBom:
+      type: array
+      items:
+            $ref: '#/components/schemas/ReturnBomItem'
+    ReturnBomItem:
```

Comment:
@gbhoot @smahre We defined an Assembly schema before, should we reuse that?

Reply:
@pkjayk The returnBom contains both Assemblies and manufacturer Items in the array. I wanted to create a schema that would work for both of those types. We could separate the two types into two different arrays possibly.

Reply:
Yes let's do that please and reuse Assembly